internal/uidriver/glfw/input.go

```diff
@@ -452,3 +453,24 @@ func standardButtonToGLFWButton(button driver.StandardGamepadButton) glfw.Gamepa
 		panic(fmt.Sprintf("glfw: invalid or inconvertible StandardGamepadButton: %d", button))
 	}
 }
+
+// UpdateStandardGamepadLayoutMappings can be used to provide new gamepad mappings to Ebiten.
+// The string must be in the format of SDL_GameControllerDB.
+func (i *Input) UpdateStandardGamepadLayoutMappings(mapping string) (bool, error) {
+	var err error
+	if i.ui.isRunning() {
+		err = u.t.Call(func() error {
+			return i.updateStandardGamepadLayoutMappings(mapping)
+		})
+	} else {
+		err = i.updateStandardGamepadLayoutMappings(mapping)
+	}
+	return err == nil, err
+}
+
+func (i *Input) updateStandardGamepadLayoutMappings(mapping string) error {
+	if !glfw.UpdateGamepadMappings(mapping) {
+		return errors.New("glfw: could not parse or update gamepad mappings")
```

Comment:
Optional: `fmt.Errorf`

Reply:
Done. Also added a comment explaining why it's a constant error string for now (go-gl's glfw API does not provide the error, that's why).